packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -1590,96 +1630,81 @@ export default abstract class SqlIntegration
-      groupByColumns,
-      eventHasUniqueTable,
-      dateFormat,
+      getAdditionalEvents,
+      getDateLimitClause,
     } = this.getSchemaFormatConfig(schemaFormat);
 
-    const today = formatDate(new Date(), dateFormat);
-    const sevenDaysAgo = formatDate(subDays(new Date(), 7), dateFormat);
-
-    const sql = `
-        SELECT
-          ${eventColumn} as event,
-          ${displayNameColumn ? displayNameColumn : eventColumn} as displayName,
-          (CASE WHEN COUNT(user_id) > 0 THEN 1 ELSE 0 END) as hasUserId,
-          COUNT (*) as count,
-          MAX(${timestampColumn}) as lastTrackedAt
-        FROM
-          ${this.generateTablePath(trackedEventTableName)}
-        WHERE
-          ${
-            eventHasUniqueTable
-              ? `received_at < '${today}' AND received_at > '${sevenDaysAgo}'`
-              : `_TABLE_SUFFIX BETWEEN '${sevenDaysAgo}' AND'${today}'`
-          } 
-        AND ${eventColumn} NOT IN ('experiment_viewed', 'experiment_started')
-        GROUP BY ${groupByColumns.join(", ")}
-    `;
+    const sql = this.getTrackedEventSql(
+      eventColumn,
+      displayNameColumn || eventColumn,
+      userIdColumn,
+      timestampColumn,
+      trackedEventTableName,
+      getDateLimitClause
+    );
 
-    const trackedResults = await this.runQuery(
+    const { rows: resultRows } = await this.runQuery(
       format(sql, this.getFormatDialect())
     );
-    const resultRows = trackedResults.rows;
-
-    if (includesPagesTable) {
-      const pageViewedSql = `
-        SELECT
-           'pages' as event,
-           "Page Viewed" as displayName,
-           (CASE WHEN COUNT(user_id) > 0 THEN 1 ELSE 0 END) as hasUserId,
-           COUNT (*) as count,
-           MAX(${timestampColumn}) as lastTrackedAt
-        FROM
-           ${this.generateTablePath("pages")}
-        WHERE received_at < '${today}' AND received_at > '${sevenDaysAgo}'`;
 
-      try {
-        const pageViewedResults = await this.runQuery(
-          format(pageViewedSql, this.getFormatDialect())
-        );
-
-        pageViewedResults.rows.forEach((row) => {
-          if (row.count > 0) {
-            resultRows.push(row);
-          }
-        });
-      } catch (e) {
-        // This happens when the table doesn't exists - this is optional, so just ignoring
-      }
-    }
-
-    if (includesScreensTable) {
-      const screenViewedSql = `
-        SELECT
-           'screens' as event,
-           "Screen Viewed" as displayName,
-           (CASE WHEN COUNT(user_id) > 0 THEN 1 ELSE 0 END) as hasUserId,
-           COUNT (*) as count,
-           MAX(${timestampColumn}) as lastTrackedAt
-        FROM
-           ${this.generateTablePath("screens")}
-        WHERE received_at < '${today}' AND received_at > '${sevenDaysAgo}'`;
+    const additionalEvents = getAdditionalEvents();
+
+    for (const additionalEvent of additionalEvents) {
+      const sql = this.getTrackedEventSql(
+        `'${additionalEvent.eventName}'`,
```

Comment:
I don't love this - but haven't found a way around it. The backtics are BigQuery specific.